youtube_dl/extractor/atresplayer.py

```diff
@@ -18,7 +18,7 @@
 
 
 class AtresPlayerIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?atresplayer\.com/television/[^/]+/[^/]+/[^/]+/(?P<id>.+?)_\d+\.html'
+    _VALID_URL = r'https?://(?:www\.)?atresplayer\.com/television/[^/]+/[^/]+/[^/]+/(?P<id>(?:(?!\1).)+)_\d+\.html'
```

Comment:
I guess you misunderstand this completely.

Simply speaking `(["\'])(?P<url>(?:(?!\1).)+)\1` regular expression means:
1. Match either `"` or `'` and remember it as `\1`.
2. Then match any character but `\1` until `\1` is found.

Thus it will match everything between quotes regardless of quotes' type (double or single) found, e.g. it will capture  `123` as `url` group of out `data-video="123"` as well as `data-video='123'` strings. This is applied in order to be future-proof in case of quotes' type change.

So it's does not make any sense here and in most of other places in this PR.